### PR TITLE
Allow measurement of line angles

### PR DESCRIPTION
**Pull request by marcjansen**
This PR enables the `measureType` angle to measure angles.

Please review.

Hey @buehner @annarieger would be nice if you could have a look.

/cc @weskamm You might be interested as well.

### CODE REVIEW

File: src/view/button/Measure.js [hunk 1]
```diff
@@ -119,72 +125,178 @@ Ext.define("BasiGX.view.button.Measure", {
...
     switchToKmOnLargeValues: true,
 
     /**
-     * determines if a marker with current measurement should be shown every
+     * Determines if a marker with current measurement should be shown every
      * time the user clicks while drawing
      */
     showMeasureInfoOnClickedPoints: false,
 
+    eventKeys: {
```
Comment (buehner) [inline src/view/button/Measure.js]: a comment for `eventKeys` would be nice :wink:

File: src/view/button/Measure.js [hunk 2]
```diff
@@ -119,72 +125,199 @@ Ext.define("BasiGX.view.button.Measure", {
...
+    /**
+     * An array of created divs we use for the tooltips. Used to eventually
+     * clean up everything we added.
+     *
+     * @type{Array<HTMLDivElement>}
+     */
+    createdTooltipDivs: [],
+
+    /**
+     * An array of created overlay we use for the tooltips. Used to eventually
```
Comment (annarieger) [inline src/view/button/Measure.js]: `overlay` -> `overlays`?

File: src/view/button/Measure.js [hunk 3]
```diff
@@ -119,72 +125,199 @@ Ext.define("BasiGX.view.button.Measure", {
...
+            me.removeMeasureTooltip();
         }
+    },
+
+    /**
+     * Called when the button is toggled, this method ensures that everything
+     * is cleaned up when unpressed, and that measuring can start when pressed.
+     *
+     * @param {Ext.Button} btn The measure button itself.
+     * @param {Boolean} pressed Whether the biutton is now pressed or not.
```
Comment (annarieger) [inline src/view/button/Measure.js]: `biutton` -> `button`

File: src/view/button/Measure.js [hunk 4]
```diff
@@ -474,5 +732,111 @@ Ext.define("BasiGX.view.button.Measure", {
...
+     * Determine the angle between two coordinates. The angle will be between
+     * -180° and 180°, with 0° being in the east. The angle will increase
+     * counter-clockwise.
+     *
+     * Inspired by http://stackoverflow.com/a/31136507
+     *
+     * @param {Array<Number>} start The start coordinates of the line with the
+     *     x-coordinate being at index `0` and y-coordinate being at index `1`.
+     * @param {Array<Number>} end The end coordinates of the line with the
+     *     x-coordinate being at index `0` and y-coordinate being at index `1`.
```
Comment (annarieger) [inline src/view/button/Measure.js]: Are indices for start and end coordinates really the same?

File: src/view/button/Measure.js (hunk 3 above)
Comment (marcjansen) [inline src/view/button/Measure.js]: Thanks, will fix

File: src/view/button/Measure.js (hunk 2 above)
Comment (marcjansen) [inline src/view/button/Measure.js]: yipp, fixing this.

**Comment by annarieger** [comment]
Very nice, @marcjansen! :+1: 
Thanks for the great documentation and tests.

**Comment by weskamm** [comment]
:+1: from me. Like the way you handle unregistering of events

**Comment by marcjansen** [comment]
Thanks @buehner, @weskamm & @annarieger for the reviews.

**Pull request merged**